src/main/java/org/kohsuke/github/GraphQLDiscussion.java

```diff
@@ -0,0 +1,202 @@
+    @SuppressFBWarnings(value = { "EI_EXPOSE_REP" }, justification = "Expected behavior")
+    private transient final GitHub root;
+
+    private Category category;
+
+    private String answerHtmlUrl;
+
+    private String answerChosenAt;
+    private GHUser answerChosenBy;
+    private String htmlUrl;
+
+    private long id;
+    private String nodeId;
+    private int number;
+    private String title;
+    private GHUser user;
+    private String state;
+    private boolean locked;
+    private int comments;
+    private String createdAt;
+    private String updatedAt;
+    private GHCommentAuthorAssociation authorAssociation;
+    private String activeLockReason;
+    private String body;
+    private String timelineUrl;
+
+    public GraphQLDiscussion() {
+        this.root = null;
+    }
+
+    public GraphQLDiscussion(GitHub root) {
+        this.root = root;
+    }
+
+    public Category getCategory() {
+        return category;
+    }
+
+    public URL getAnswerHtmlUrl() {
+        return GitHubClient.parseURL(answerHtmlUrl);
+    }
+
+    public Date getAnswerChosenAt() {
+        return GitHubClient.parseDate(answerChosenAt);
+    }
+
+    public GHUser getAnswerChosenBy() throws IOException {
+        return root().intern(answerChosenBy);
+    }
+
+    public URL getHtmlUrl() {
+        return GitHubClient.parseURL(htmlUrl);
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public String getNodeId() {
+        return nodeId;
+    }
+
+    public int getNumber() {
+        return number;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public GHUser getUser() throws IOException {
+        return root().intern(user);
+    }
+
+    public State getState() {
+        return EnumUtils.getEnumOrDefault(State.class, state, State.UNKNOWN);
+    }
+
+    public boolean isLocked() {
+        return locked;
+    }
+
+    public int getComments() {
+        return comments;
+    }
+
+    public Date getCreatedAt() {
+        return GitHubClient.parseDate(createdAt);
+    }
+
+    public Date getUpdatedAt() {
+        return GitHubClient.parseDate(updatedAt);
+    }
+
+    public GHCommentAuthorAssociation getAuthorAssociation() {
+        return authorAssociation;
+    }
+
+    public String getActiveLockReason() {
+        return activeLockReason;
+    }
+
+    public String getBody() {
+        return body;
+    }
+
+    public String getTimelineUrl() {
+        return timelineUrl;
+    }
+
+    /**
+     * Get the root {@link GitHub} instance for this object.
+     *
+     * @return the root {@link GitHub} instance
+     */
+    @NonNull
+    GitHub root() {
+        return Objects.requireNonNull(root,
+                "The root GitHub reference for this instance is null. Probably caused by deserializing this class without using a GitHub instance. If you must do this, use the MappingObjectReader from GitHub.getMappingObjectReader().");
+    }
+
+    public static class Category {
```

Comment:
https://docs.github.com/en/graphql/guides/using-the-graphql-api-for-discussions#discussioncategory

Reply:
Did you add this comment for me to add a javadoc? Or do you want me to adjust the name?

Reply:
Just add JavaDoc, but may also note that GHRepositoryDiscussion (and nested classes) are not exactly the same their GraphQL counterparts.